examples/with-ecs-scheduling/main.tf

```diff
@@ -99,37 +95,45 @@ module "eventbridge" {
 # ECS
 ######
 
-module "ecs" {
-  source  = "terraform-aws-modules/ecs/aws"
-  version = "~> 3.0"
+module "ecs_cluster" {
+  source  = "terraform-aws-modules/ecs/aws//modules/cluster"
+  version = "~> 5.0"
 
-  name = random_pet.this.id
+  cluster_name = random_pet.this.id
 
-  capacity_providers = ["FARGATE", "FARGATE_SPOT"]
+  fargate_capacity_providers = {
+    FARGATE = {
+      default_capacity_provider_strategy = {
+        weight = 100
+      }
+    }
+    FARGATE_SPOT = {
+      default_capacity_provider_strategy = {
+        weight = 100
+      }
+    }
+  }
 }
 
-resource "aws_ecs_service" "hello_world" {
-  name            = "hello_world-${random_pet.this.id}"
-  cluster         = module.ecs.ecs_cluster_id
-  task_definition = aws_ecs_task_definition.hello_world.arn
+module "ecs_service" {
```

Comment:
Btw, you can call the root ECS module instead of two separate submodules (cluster and service) for the brevity of this example. See this example - https://github.com/terraform-aws-modules/terraform-aws-ecs#integrated-cluster-w-services